addons/mass_mailing/controllers/main.py

```diff
@@ -195,3 +195,14 @@ def send_feedback(self, mailing_id, res_id, email, feedback, token):
                 record.sudo().message_post(body=_("Feedback from %(email)s: %(feedback)s", email=email, feedback=feedback))
             return bool(records)
         return 'error'
+
+    @http.route('/mailing/report/unsubscribe', type='http', website=True, auth='public')
+    def turn_off_mailing_reports(self, token, user_id):
+        if not token or not user_id:
+            raise werkzeug.exceptions.NotFound()
+        correct_token = consteq(token, request.env['mailing.mailing']._get_unsubscribe_token(user_id))
+        user = request.env['res.users'].sudo().browse(user_id)
```

Comment:
does it work like this ? (not sure if you need to do `int(user_id)` because the parameter should be a string)